### PR TITLE
Disallow multiply defined coordinate reference parameters when writing

### PR DESCRIPTION
**Pull request by davidhassell**
Fixes: #148

### CODE REVIEW

**Comment by davidhassell** [comment]
Hi Sadie, it'd great if you could review this short PR (the changes from https://github.com/NCAS-CMS/cfdm/pull/150/commits/825feb192c74bc86068c71a03b64d64c8b3bd06a are  a bit messy, but are just bringing the style up to date).

Thanks

File: Changelog.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ version 1.8.?.?
 
 * Construct access API changes from 1.8.9.0 applied to `Field.convert`
 * Improved error message for invalid inputs to `Field.convert`
+* Raise exception when attempting to write multiply defined coordinate
+  reference parameters(https://github.com/NCAS-CMS/cfdm/issues/148)
```
Comment (sadielbartholomew) [inline Changelog.rst:10]:
```suggestion
  reference parameters (https://github.com/NCAS-CMS/cfdm/issues/148)
```

File: cfdm/read_write/netcdf/netcdfwrite.py [hunk 2]
```diff
@@ -2560,6 +2536,15 @@ def _write_grid_mapping(self, f, ref, multiple_grid_mappings):
 
             # Add named parameters
             parameters = self.implementation.get_datum_parameters(ref)
+
+            common = set(parameters).intersection(cc_parameters)
+            if common:
+                raise ValueError(
+                    "Can't create CF-netDF grid mapping variable: "
```
Comment (sadielbartholomew) [inline cfdm/read_write/netcdf/netcdfwrite.py:2543]:
```suggestion
                    "Can't create CF-netCDF grid mapping variable: "
```

**Review by sadielbartholomew — APPROVED** [review]
Addresses and tests for the issue at hand; with the new `common` logic block removed the new test fails. The added error message is concise and clear. Two small typos notes, otherwise this is good to merge in my eyes.

**Comment by davidhassell** [comment]
Thanks, Sadie - don't know how I missed those conflicts! Merging now ...

**Pull request merged**